ttnn/cpp/ttnn/operations/moreh/moreh_adam/device/moreh_adam_device_operation.cpp

```diff
@@ -156,4 +156,14 @@ std::tuple<MorehAdamOperation::operation_attributes_t, MorehAdamOperation::tenso
             max_exp_avg_sq_in,
             {param_out, exp_avg_out, exp_avg_sq_out, max_exp_avg_sq_out}}};
 }
+auto MorehAdamOperation::compute_program_hash(
+    const MorehAdamOperation::operation_attributes_t& operation_attributes,
+    const MorehAdamOperation::tensor_args_t& tensor_args) -> tt::stl::hash::hash_t {
+    // For hash we'll set `step` to 0
+    auto operation_attributes_without_step = operation_attributes;
+    operation_attributes_without_step.step = 0;
```

Comment:
wdyt about not caching `learning_rate` as well? Once we start using LRScheduler we will same issue.

Reply:
I think 305e5fd should cover it. You should double check if the rewrite of `compute_program_hash` is acceptable as well